internal/cmd/api-key/command_delete.go

```diff
@@ -28,17 +25,42 @@ func (c *command) newDeleteCommand() *cobra.Command {
 
 func (c *command) delete(cmd *cobra.Command, args []string) error {
 	c.setKeyStoreIfNil()
-	apiKey := args[0]
 
-	promptMsg := fmt.Sprintf(errors.DeleteResourceConfirmYesNoMsg, resource.ApiKey, apiKey)
-	if ok, err := form.ConfirmDeletion(cmd, promptMsg, ""); err != nil || !ok {
+	if confirm, err := c.confirmDeletion(cmd, args); err != nil {
 		return err
+	} else if !confirm {
+		return nil
+	}
+
+	deleteFunc := func(id string) error {
+		if r, err := c.V2Client.DeleteApiKey(id); err != nil {
+			return errors.CatchApiKeyForbiddenAccessError(err, deleteOperation, r)
+		}
+		return nil
 	}
-	httpResp, err := c.V2Client.DeleteApiKey(apiKey)
+
+	deleted, err := resource.Delete(args, deleteFunc, c.postProcess)
```

Comment:
Why do we need to pass a post-process function? Can't we just call it afterwards?

Reply:
I passed it in so that I wouldn't have to loop over the id's a second time.

If it improves readability, I can make it take the list of successfully deleted resources as an argument and call it outside.

Reply:
Yeah, I think that would help. We should 1) avoid passing functions into functions where possible and 2) avoid non-descriptive terminology like "postProcess"